`No module named 'distutils'` running tests under Python 3.12

# To reproduce:

1. Under Python 3.12 run `./run_tests.sh`

# Result:

```
__________________________________ ERROR collecting tests/mrjob/test_mrjob.py ___________________________________
ImportError while importing test module '/Users/jaybazuzi/source/ApprovalTests_Python/tests/mrjob/test_mrjob.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/mrjob/test_mrjob.py:1: in <module>
    from mrjob.job import MRJob
.tox/py/lib/python3.12/site-packages/mrjob/job.py:36: in <module>
    from mrjob.conf import combine_dicts
.tox/py/lib/python3.12/site-packages/mrjob/conf.py:34: in <module>
    from mrjob.util import expand_path
.tox/py/lib/python3.12/site-packages/mrjob/util.py:31: in <module>
    from distutils.spawn import find_executable
E   ModuleNotFoundError: No module named 'distutils'
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -9,7 +9,7 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        python-version: [3.8, 3.9, "3.10", "3.11"]
+        python-version: [3.8, 3.9, "3.10", "3.11", "3.12"]
         os: [macos-latest, ubuntu-latest, windows-latest]
 
     steps:

--- a/requirements.test.txt
+++ b/requirements.test.txt
@@ -1,3 +1,4 @@
+setuptools
 numpy
 pylint
 pytest-asyncio==0.21.1 # 0.23 has a bug 2023/12/3

